### ISSUE TITLE
Update all remaining legacy SQL queries to use the new syntax

### ISSUE THREAD

**Issue by kalbfled** [department-of-veterans-affairs/notification-api#1648]
## User Story - Business Need

SQLAlchemy v1.4 supporters two syntaxes for making database queries, but SQLAlchemy v2 only supports the newer syntax.  Before we can update to v2, we need to convert all instances of legacy syntax to the newer syntax.

This work began with #1538.  This ticket should finish the job.

### User Story(ies)

**As a** VA Notify developer
**I want** to update legacy database query syntax to the new version
**So that** we can upgrade to SQLAlchemy v2.

- [x] discuss w/ QA

### Additional Info and Resources

- [SQLAlchemy v2.x changelog](https://docs.sqlalchemy.org/en/20/changelog/changelog_20.html)
- [Migration Guide](https://docs.sqlalchemy.org/en/20/changelog/migration_20.html)

app queries
<img width="595" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeVJqQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--44a51d77f13e08846ab41fbf70d105f151781df3/image.png" alt="image.png" />

## Engineering Checklist

- [x] Review relevant upgrade/migration docs before getting started

**IMPORTANT NOTE:** Legacy queries generally start with `<model class>.query`.  Depending on context, updated syntax might use a new style query (i.e. `stmt = select(Notification).where(...)` followed by an execution statement) or an updated ORM call (i.e. replace `Notification.query.count() == 1` with `notify_db_session.session.get(Notification, notification_id) is not None`, or write a statement to select the count).

- [x] Refactor all query statements the match these regular expressions, which overlap significantly:
  - [x] `grep -rn -E "\.query(\.|\()"`
  - [x] `grep -rn filter_by` (not all matches are part of queries)
- [x] Update the following `./app/` queries:
  - [x] app/dao/complaint_dao.py
  - [x] app/dao/services_dao.py
  - [ ] The remaining `./app/` queries need a `TODO:` comment and to be deleted, ensuring the methods return appropriate values.
    - [ ] `TODO` comments have tickets listed. These should be cleanup tickets. We do not use fido2, organisations, nor letters.
- [x] Collect unit tests that need updating and/or investigation into a new ticket. **No changes to tests are required for this task!**
  - We sometimes see unit tests that query SMS and email but only add notifications for one or the other. Or there are multiple services queried but the test only does one. Add a `TODO` comment and add these to a ticket, so we can investigate them. 
 
## Acceptance Criteria

- [x] No legacy queries remain
- [x] All unit tests pass

## QA Considerations
- [x] QA regression passes
- [x] Pay extra care to JSON responses
- [x] Performance Testing to ensure there are no significant slowdowns to investigate

## Potential Dependencies
_Leave blank if n/a_

## Out of Scope

- Do not upgrade any dependency.  Separate tickets exist for those upgrades.
- Do not update unit tests as part of this

**Comment by mjones-oddball** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/VA-Notify-620d21369d810a00146ed9c8/issues/department-of-veterans-affairs/notification-api/1648?planning-poker) with Zenhub @cris-oddball @EvanParish @k-macmillan @ldraney

**Comment by kalbfled** [comment]
I upgrade all remaining legacy queries in app/ as part of the first task (the grep statements).  Ergo, the TODO task became a non-issue.

**Comment by cris-oddball** [comment]
Needing @kalbfled to check the AC boxes off in this ticket. Will look at performance on Monday.
